### DISCUSSION TITLE
401 authentication error - read/write on from twitter dashboard but problem persisting

### DISCUSSION THREAD

**Discussion by bybilal** [tweepy/tweepy#1780] [Questions]
I want to call the client.like method in tweepy after Im attempting to properly authenticate with ouath1a. Can someone help me please, thanks:

i think this is the endpoint via twitter that tweepy is calling?

POST favorites/create

and here’s the method in tweepy:
tweepy.Client — Twitter API v2 Reference — tweepy 4.4.0 documentation that’s the method

thanks for the advice, here’s my codebase:
`
I want to call the **client.like** method in *tweepy* after I've properly authenticated with ouath1a i guess?

i think this is the endpoint via twitter that tweepy is calling?
 
**POST favorites/create**

and here's the method in tweepy:
 https://docs.tweepy.org/en/stable/client.html that's the method

thanks for the advice, here's my codebase:

```
import tweepy
import config
import time

# creating client returning client
def getClient():
    x = tweepy.Client(bearer_token=config.bearer_token,

                       consumer_key=config.consumer_key,
                       consumer_secret=config.consumer_secret,
                       access_token=config.access_token,
                       access_token_secret=config.access_token_secret
                       ) 
    return x

# testing if i can get full user info
def getUserInfo():
    y = getClient()
    user = y.get_user(username='davechappelle')
    return user.data

user = getUserInfo()


print(user)
 
```

In the docs it says: Since you are making requests on behalf of a user, you must authenticate these endpoints with [OAuth 1.0a User Context](https://developer.twitter.com/en/docs/authentication/oauth-1-0a) and use the Access Tokens associated with the user, which can be generated using the[ 3-legged OAuth flow](https://developer.twitter.com/content/developer-twitter/en/docs/authentication/oauth-1-0a/obtaining-user-access-tokens).

maybe that's what i need to do ^ but isn't that what my code base is doing? thanks

**Comment by Harmon758** [comment]
Check that your credentials are correct. If necessary, try regenerating them.
